packages/cli/src/Server.ts

```diff
@@ -1544,61 +1547,71 @@ class App {
 		// Creates new credentials
 		this.app.post(
 			`/${this.restEndpoint}/credentials`,
-			ResponseHelper.send(
-				async (req: express.Request, res: express.Response): Promise<ICredentialsResponse> => {
-					const incomingData = req.body;
+			ResponseHelper.send(async (req: CredentialRequest.Create) => {
+				delete req.body.id; // delete if sent
 
-					if (!incomingData.name || incomingData.name.length < 3) {
-						throw new ResponseHelper.ResponseError(
-							`Credentials name must be at least 3 characters long.`,
-							undefined,
-							400,
-						);
-					}
+				const newCredential = new CredentialsEntity();
 
-					// Add the added date for node access permissions
-					for (const nodeAccess of incomingData.nodesAccess) {
-						nodeAccess.date = this.getCurrentDate();
-					}
+				Object.assign(newCredential, req.body);
 
-					const encryptionKey = await UserSettings.getEncryptionKey();
-					if (encryptionKey === undefined) {
-						throw new Error('No encryption key got found to encrypt the credentials!');
-					}
+				await validateEntity(newCredential);
 
-					if (incomingData.name === '') {
-						throw new Error('Credentials have to have a name set!');
-					}
+				// Add the added date for node access permissions
+				for (const nodeAccess of newCredential.nodesAccess) {
+					nodeAccess.date = this.getCurrentDate();
+				}
 
-					// Encrypt the data
-					const credentials = new Credentials(
-						{ id: null, name: incomingData.name },
-						incomingData.type,
-						incomingData.nodesAccess,
-					);
-					credentials.setData(incomingData.data, encryptionKey);
-					const newCredentialsData = credentials.getDataToSave() as ICredentialsDb;
+				const encryptionKey = await UserSettings.getEncryptionKey();
 
-					await this.externalHooks.run('credentials.create', [newCredentialsData]);
+				if (!encryptionKey) {
+					throw new Error('No encryption key was found to encrypt the credential!');
+				}
```

Comment:
This is the stub from the original endpoint - I thought I'd removed it but I kept it.  `GET /credentials` does not [return data](https://mattermost.internal.n8n.io/n8n/pl/7fiso6dp8pydzxdzytgm5961rc), so it looks safe to remove. Let me know.